src/actions/smartWalletActions.js

```diff
@@ -1208,14 +1208,16 @@ export const fetchAvailableTxToSettleAction = () => {
     const activeAccountAddress = activeAccountAddressSelector(getState());
     const accountEthereumHistory = archanovaAccountEthereumHistorySelector(getState());
     const accountAssets = archanovaAccountEthereumAssetsSelector(getState());
+    const accountAssetsList = getAssetsAsList(accountAssets);
```

Comment:
The longer I think about our wallet assets balances and supportes assets, I think they should be otherway around:

`walletAssetBalances` should be array as a simple list of user assets
`supportedAssets` should be record by address to ease up lookup.

That's just my feeling, I know that changing this would be huge undertaking ;-)

Reply:
I does makes sense. But as you stated it's one more huge refactor so for this task let's leave as it is and keep the improvement in mind. Sounds good?